### PR TITLE
Grupozap webhook

### PR DESCRIPTION
**Pull request by rhnonose**
- [x] Plug accepting `POST` with basic auth
- [x] Simple persistence
- [ ] Associate listing
- [ ] Associate user
- [ ] Send e-mail

Since we need to push to staging so they can test our API, I might leave the rest of the check-list for another PR.

### CODE REVIEW

File: apps/re/lib/leads/grupozap_buyer.ex [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  """
+  use Ecto.Schema
+
+  import Ecto.Changeset
+
+  @primary_key {:uuid, :binary_id, autogenerate: false}
+
+  schema "grupozap_buyer_leads" do
+    field :lead_origin, :string
+    field :timestamp, :utc_datetime
```
Comment (joaodubas) [inline apps/re/lib/leads/grupozap_buyer.ex]: \o/

File: apps/re/lib/leads/grupozap_buyer.ex [hunk 2]
```diff
@@ -0,0 +1,36 @@
+defmodule Re.Leads.GrupozapBuyer do
```
Comment (joaodubas) [inline apps/re/lib/leads/grupozap_buyer.ex]: Will we separate each lead source in a different table?

Comment (rhnonose) [reply apps/re/lib/leads/grupozap_buyer.ex]: Not sure yet.
I'm creating just a "mirror" that'll save the POST from the integration and then async jobs to create a "clean lead", we can decide later if it's a single model or not.

File: apps/re_web/test/webhooks/grupozap_plug_test.exs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    Leads.GrupozapBuyer,
+    Repo
+  }
+
+  @payload %{
+      "leadOrigin" => "VivaReal",
+      "timestamp" => "2019-01-01T00:00:00.000Z",
+      "originLeadId" => "59ee0fc6e4b043e1b2a6d863",
+      "originListingId" => "87027856",
+      "clientListingId" => "a40171",
```
[the remark this answers is not in the document]
Comment (rhnonose) [reply apps/re_web/test/webhooks/grupozap_plug_test.exs]: Yikes.
Is there any documentation on the format? Can we use listing's `uuid`?

File: apps/re_web/test/webhooks/grupozap_plug_test.exs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  describe "POST" do
+    test "authenticated request", %{authenticated_conn: conn} do
+      conn = post(conn, "/webhooks/grupozap", @payload)
+
+      assert text_response(conn, 200) == "ok"
+
+      assert Repo.one(GrupozapBuyer)
+    end
+
+    @tag capture_log: true
```
[the remark this answers is not in the document]
Comment (rhnonose) [reply apps/re_web/test/webhooks/grupozap_plug_test.exs]: Not, this test logs an `error`, so it's to keep the testing log clean.

File: apps/re_web/test/webhooks/grupozap_plug_test.exs (hunk 3 above)
[the remark this answers is not in the document]
Comment (joaodubas) [reply apps/re_web/test/webhooks/grupozap_plug_test.exs]: We send our ID in all integrations:

1. ZAP: https://github.com/emcasa/backend/blob/43d653fa93faa75ec1454085d8f057f74a010bad/apps/re/lib/exporters/zap.ex#L44-L46
2. VivaReal: https://github.com/emcasa/backend/blob/43d653fa93faa75ec1454085d8f057f74a010bad/apps/re/lib/exporters/vivareal.ex#L58-L60
3. ImovelWeb: https://github.com/emcasa/backend/blob/43d653fa93faa75ec1454085d8f057f74a010bad/apps/re/lib/exporters/imovelweb.ex#L47
4. Trovit: https://github.com/emcasa/backend/blob/43d653fa93faa75ec1454085d8f057f74a010bad/apps/re/lib/exporters/trovit.ex#L52-L54

[the remark this answers is not in the document]
Comment (joaodubas) [reply apps/re_web/test/webhooks/grupozap_plug_test.exs]: I think they only accept integers in this field, but we can double check their documentation.

[the remark this answers is not in the document]
Comment (rhnonose) [reply apps/re_web/test/webhooks/grupozap_plug_test.exs]: The integration mentions explicitly `externalId`, so we might have to make a copy.
And using our internal ID to reference outside is not a good idea, ideally we should use `uuid`.

**Pull request merged**